org.eclipse.xtext.util/src/org/eclipse/xtext/util/MergeableManifest2.java

```diff
@@ -0,0 +1,652 @@
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.util;
+
+import java.io.BufferedReader;
+import java.io.BufferedWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.OutputStream;
+import java.io.OutputStreamWriter;
+import java.util.Collection;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.Map.Entry;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+/**
+ * Functionality to read and write Manifest files. Guarantees that modifying
+ * does not destroy the order of the elements to ensure minimal diffs for
+ * load/save cycles.
+ * 
+ * Re-implementation from {@link MergeableManifest} but without reflection.
+ * Reflection causes warnings that it might get be removed in future versions.
+ * Hence we implement the functionality with all the inherited one by ourself.
+ * 
+ * @author Arne Deutsch - Initial contribution and API
+ */
+public class MergeableManifest2 implements Cloneable {
+
+	public static final String MANIFEST_VERSION = "Manifest-Version";
+	public static final String SIGNATURE_VERSION = "Signature-Version";
+	public static final String REQUIRE_BUNDLE = "Require-Bundle";
+	public static final String BUNDLE_SYMBOLIC_NAME = "Bundle-SymbolicName";
+	public static final String EXPORT_PACKAGE = "Export-Package";
+	public static final String IMPORT_PACKAGE = "Import-Package";
+	public static final String BUNDLE_REQUIREDEXECUTIONENVIRONMENT = "Bundle-RequiredExecutionEnvironment";
+	public static final String BUNDLE_ACTIVATOR = "Bundle-Activator";
+
+	private String name;
+	private String version;
+	private String bree;
+	private String bundleActivator;
+	private String newline = System.lineSeparator();
+	private Attributes mainAttributes = new Attributes();
+	private Map<String, Attributes> entries = new LinkedHashMap<>();
+	private boolean modified;
+
+	public MergeableManifest2(InputStream stream, String name) throws IOException {
+		this.name = name;
+		read(stream);
+		this.modified = false;
+	}
+
+	public MergeableManifest2(InputStream stream) throws IOException {
+		read(stream);
+		this.modified = false;
+	}
+
+	public MergeableManifest2(MergeableManifest2 toCopy) {
+		this.name = toCopy.name;
+		this.version = toCopy.version;
+		this.newline = toCopy.newline;
+		this.mainAttributes.putAll(toCopy.mainAttributes);
+		this.entries.putAll(toCopy.entries);
+		this.modified = false;
+	}
+
+	private void read(InputStream stream) throws IOException {
+		read(new BufferedReader(new InputStreamReader(stream)).lines().collect(Collectors.toList()));
```

Comment:
Please document if the stream will be closed or not after an invocation of read(..)

Reply:
Will enhance documentation.